Mage/src/main/java/mage/cards/CardsImpl.java

```diff
@@ -134,9 +130,14 @@ public Set<Card> getCards(FilterCard filter, UUID playerId, Ability source, Game
         return cards;
     }
 
+    // TODO: Why is this used a completely different implementation than the version without the filter?
```

Comment:
I think it's a miss fix here. Must be same code as `getCards` without filter (single code for both), e.g. with permanent's fix (first introduced in 14b2760e99f24c9ddbcdf2cf5f31337bbf69ee92 with related test).

Reply:
I think it would be good to merge this to improve documentation and I don't see any other problems with it. Is it okay to merge as is (after checking conflicts) and open a separate issue for this todo?